cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -135,6 +135,69 @@ using rhs_iterator = strong_index_iterator<rhs_index_type>;
 
 namespace lexicographic {
 
+/**
+ * @brief Relational comparator functor that evaluates `NaN` as not less, not equivalent, and not
+ * greater than all other values. This is IEE 754 compliant.
+ */
+struct physical_element_comparator {
+  /**
+   * @brief Shared operator for all `Element` types relational comparisons.
+   *
+   * @param[in] lhs first element
+   * @param[in] rhs second element
+   * @return Indicates the relationship between the elements in
+   * the `lhs` and `rhs` columns.
+   */
+  template <typename Element>
+  __device__ weak_ordering operator()(Element lhs, Element rhs)
+  {
+    return detail::compare_elements(lhs, rhs);
+  }
+};
+
+/**
+ * @brief Relational comparator functor that evaluates `NaN` as equivalent to other `NaN`s and
+ * greater than all other values.
+ */
+struct sorting_physical_element_comparator : physical_element_comparator {
```

Comment:
So this inheritance ends up being meaningless because the two `operator()` overloads end up hiding anything inherited from the base.

Reply:
Thanks, had tried to split the operator in `physical_element_comparator` and override just the floating point specialization, but ended up with compile errors because it wanted a complete definition.